packages/graphql-hooks/src/GraphQLClient.ts

```diff
@@ -425,8 +425,11 @@ class GraphQLClient {
     }
   }
 
-  setQueryData(query: string, updater: (oldState?: any) => any): void {
-    const cacheKey = this.getCacheKey({ query })
+  setQueryData(query: string, updater: (oldState?: any) => any, operation: Operation | {} = {}): void {
```

Comment:
let's make the query argument dynamic so that if it's a string we assume it's the query, if it's an object we treat it as an operation. let's also update the invalidatQuery method to do the same.